typing/typecore.ml

```diff
@@ -3862,7 +3868,10 @@ and type_binding_op_ident env s =
   in
   path, desc
 
-and type_function ?in_function loc attrs env ty_expected_explained l caselist =
+and type_function : ?in_function: Location.t * type_expr ->
+  Location.t -> Parsetree.attributes -> Env.t -> type_expected -> arg_label ->
+  Parsetree.case list -> expression
+= fun ?in_function loc attrs env ty_expected_explained l caselist ->
```

Comment:
I think that diff is more noise than anything.
Perhaps just a type annotation on the `?in_function` parameter and a better name for `l` would be enough?

Reply:
Certainly those two arguments are the ones difficult to guess their types.
`arg_label` instead of `l` might be fine?

Reply:
Yes, I think `arg_label` would make sense :)